### ISSUE TITLE
limit number of ACK frames per received packets to 1

### ISSUE THREAD

**Issue by marten-seemann** [lucas-clemente/quic-go#247]
This makes it a little bit more difficult to DOS us with sending us ACKs, since the attacker has to send a whole packet per ACK (thus creating overhead for PublicHeader and crypto).

**Comment by marten-seemann** [comment]
Maybe we should limit the number to 2 or 3 (as a server parameter), such that a stupid (future) implementation doesn't run into a `ConnectionClose` that easily.
A possible case where this might happen is when packets are retransmitted, without stripping ACK frames from the packet.

@lucas-clemente: What do you think?

**Comment by lucas-clemente** [comment]
I would prefer restricting it to 1 and emitting a helpful error message (`found more than one ACK frame in a packet`). I expect that will help a future implementer more than silently accepting nonsensical packets.

**Comment by marten-seemann** [comment]
Well, that only helps with clients that are tested against our implementation. Other implementations might be incompatible with quic-go since Google's QUIC server implementation doesn't enforce this.

Since we now fixed all security-related bugs in the `AckHandler`, there's not a lot to gain by enforcing this restriction anyway. I'd be happy to reopen this issue once Chrome enforces this, or once the spec mandates it. Closing the issue for now.
